File: Csharp/CodeWars/src/Resolutions/RegexpBasicsIsItADigit.cs
using System.Text.RegularExpressions;

namespace CodeWars.src.Resolutions;

public static class RegexpBasicsIsItADigit
{
    public static bool Digit(this string s) =>
        s.Length == 1
        && Regex.Replace(s, "[0-9]", "").Count() == 0;
}

File: TypeScript/Codewars/src/resolutions/Regexp-Basics-is-it-a-digit.ts
interface String {
	digit(): boolean;
}

String.prototype.digit = function (this: string): boolean {
	return this.length === 1 && /^[0-9]$/.test(this);
};

console.log("".digit(), false);
console.log("7".digit(), true);
console.log(" ".digit(), false);
console.log("a".digit(), false);
console.log("a5".digit(), false);
console.log("14".digit(), false);
